### ISSUE TITLE
Missing completion for crane package

### ISSUE THREAD

**Issue by sestegra** [NixOS/nixpkgs#310606]
### Describe the bug
After installation crane@0.19.1, shell completion is not available.

### Steps To Reproduce
1. Install crane@0.19.1
2. Try completion

### Expected behavior
Shell completion should be installed.

### Screenshots
N/A

### Additional context
N/A

### Notify maintainers

@yurrriq

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.0-105-generic, Ubuntu, 22.04.4 LTS (Jammy Jellyfish), nobuild`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/home/sestegra/.nix-defexpr/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by yurrriq** [comment]
Thanks for the report! I'd gladly review a PR if you want to submit one and @ me.

**Comment by yurrriq** [comment]
I went ahead and created https://github.com/NixOS/nixpkgs/pull/322751. Might you able to test? Thanks again for calling this out!

**Comment by yurrriq** [comment]
I managed to verify it and I'm quite confident it'll work for other systems and shells.

**Comment by joneshf** [comment]
I spent a decent amount of time trying to figure out why I didn't have completions, so to hopefully save others from running into a similar issue…Turns out, the [`crane` package](https://search.nixos.org/packages?channel=unstable&show=crane) is only the [`crane` output](https://github.com/NixOS/nixpkgs/blob/08f22084e6085d19bcfb4be30d1ca76ecb96fe54/pkgs/by-name/go/go-containerregistry/package.nix#L34) of the derivation, which doesn't include the other stuff like the completions. The [`go-containerregistry` package](https://search.nixos.org/packages?channel=unstable&show=go-containerregistry) is the package with the `out` output that includes both the `crane` binary and the other stuff, like the completions.

**tl;dr;** use `go-containerregistry` if you want `crane` and its completions.

**Comment by yurrriq** [comment]
Hmm, should we change `outputs` somehow? When I `nix build github:nixos/nixpkgs#go-containerregistry` I get the following.

```
result
├── bin
│   ├── crane -> /nix/store/gap8lddzsinipk0lfjg4sxgahkp3wxsr-go-containerregistry-0.20.6-crane/bin/crane
│   └── gcrane -> /nix/store/fbgv94yh8mxnzp47vi52dzwvr7b3j5g3-go-containerregistry-0.20.6-gcrane/bin/gcrane
└── share
    ├── bash-completion
    │   └── completions
    │       ├── crane.bash
    │       └── gcrane.bash
    ├── fish
    │   └── vendor_completions.d
    │       ├── crane.fish
    │       └── gcrane.fish
    └── zsh
        └── site-functions
            ├── _crane
            └── _gcrane
```

and indeed for `crane` I get:

```
result-crane/
└── bin
    └── crane
```

**Comment by joneshf** [comment]
Sorry, I'm not actually familiar with how `outputs` is supposed to be used. This might be the right way to use them, I honestly don't know.
